src/utils/utils.ts

```diff
@@ -426,3 +426,36 @@ export const isValidProtocol = (protocolAddress: string): boolean => {
   );
   return validProtocolAddresses.includes(checkSumAddress);
 };
+
+/**
+ * Decode an encoded list of token ids
+ * @param encodedTokenIds encoded token ids from the criteria of an order
+ * @returns a list of token ids
+ */
+export const decodeTokenIds = (encodedTokenIds: string): string[] => {
+  if (encodedTokenIds.length === 0) {
+    return [];
+  }
+
+  if (encodedTokenIds == "*") {
+    return ["*"];
+  }
+
+  const ranges = encodedTokenIds.split(",");
+  const numbers: string[] = [];
+
+  for (const r of ranges) {
+    if (r.includes(":")) {
+      const [start, end] = r.split(":").map(Number);
+      numbers.push(
+        ...Array.from({ length: end - start + 1 }, (_, i) =>
```

Comment:
nice es6 solution, although would recommend we make a little util fn and import, then it should be really nice and concise to read here

```typescript
const range = (start: number, end: number) => Array.from(
  Array(Math.abs(end - start) + 1), 
  (_, i) => start + i
);
```

Reply:
actually this is not going to work for big numbers, you'll have to use BigNumber in case the token ids are past the max safe js number

Reply:
you'll want something like this

```typescript
const [start, end] = r.split(":").map(BigNumber.from);
```

```typescript
const range = (start: BigNumber, end: BigNumber) => Array.from(
  Array(end.sub(start).add(1).toNumber()), 
  (_, i) => start.add(i).toNumber()
);
```